app/controllers/favourites_controller.rb

```diff
@@ -0,0 +1,22 @@
+class FavouritesController < ApplicationController
+  before_action :authenticate_user!
+
+  def index
+      @posts = Post.all
+      @favourites = current_user.favourites
+  end
+
+  def create
+    @post = Post.find params[:post_id]
+    f = Favourite.create(post: @post, user: current_user)
+    redirect_to post_path(@post), notice: "you have favourited this post"
+  end
+
+  def destroy
+    favourite = Favourite.find params[:id]
+    post = Post.find params[:post_id]
+    favourite.destroy if can? :destroy, Favourite
```

Comment:
The favourite here doesn't need to be a constant, just a normal variable as it is defined earlier in the method.